commcare_connect/opportunity/visit_import.py

```diff
@@ -478,6 +479,8 @@ def _bulk_update_catchments(opportunity: Opportunity, dataset: Dataset):
                 invalid_rows.append((row, f"Error in row {row}: {e}"))
 
         if to_create:
+            for ca in to_create:
+                ca.site_code = slugify_uniquely(ca.name, CatchmentArea, "site_code")
```

Comment:
if the users specifies a site code, that should be used instead of just slug of the name